cluster/points_writer.go

```diff
@@ -196,6 +202,13 @@ func (w *PointsWriter) WritePoints(database, retentionPolicy string, consistency
 	w.statMap.Add(statWriteReq, 1)
 	w.statMap.Add(statPointWriteReq, int64(len(points)))
 
+	if w.MaxConcurrentWrites > 0 {
```

Comment:
I was actually saying that there would be two levels. The query executor would have a maximum number of queries that it would run that the query executor would enforce while the httpd service would have its own enforcement to prevent DDOS attacks. One would act on the listener while the other would act on the service and would have different performance characteristics and error messages.

Although I'm not sure if both would be needed in all honesty. I think just a single connection limit on the httpd listener would probably work best.